trace/cloud-trace-demo-app-opentelemetry/app/app.py

```diff
@@ -40,20 +47,40 @@ def configure_exporter(exporter):
 
 configure_exporter(CloudTraceSpanExporter())
 tracer = trace.get_tracer(__name__)
+# [END trace_demo_create_exporter]
 
 
 @app.route("/")
 def template_test():
     # Sleep for a random time to imitate a random processing time
     time.sleep(random.uniform(0, 0.5))
 
-    with tracer.start_as_current_span("span1"):
-        with tracer.start_as_current_span("span2"):
-            with tracer.start_as_current_span("span3"):
-                print("Hello world from Cloud Trace Exporter!")
+    # Keyword that gets passed in will be concatenated to the final output string.
+    keyword = app.config['keyword']
+    # If there is no endpoint, return the output string.
+    url = app.config['endpoint']
+    if url == "":
+        return keyword, 200
+
+    # Endpoint is the next service to send string to.
+    data = {'body': keyword}
 
-    return "Hello World"
+    # [START trace_context_header]
+    with tracer.start_as_current_span("span_" + keyword):
+        response = requests.get(
```

Comment:
This call will not properly propagate the trace context as is. It looks like the OC app.py was manually adding the header. Would you prefer to

1. Manually pass the header here as well (I can help you with the propagator code)
2. Use the [`opentelemetry-instrumentation-requests`](https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/requests/requests.html) package (similar to the `FlaskInstrumentor()`) to get propagation. This will also create a client side span for the request.

I think 2 is the way to go.

Reply:
Thank you. IMO, Option 2 is best here both for implementation and for demonstrating OT instrumentation.

Reply:
@aabmass looking at the sample in the [documentation][doc] the instrumentor is initialized.
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/9a7861952f0a42dbe3195b8b2b940fb1d9549c30/trace/cloud-trace-demo-app-opentelemetry/app/app.py#L36-L37
Do you mean that the following call to `tracer.start_as_current_span()` should be removed?
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/9a7861952f0a42dbe3195b8b2b940fb1d9549c30/trace/cloud-trace-demo-app-opentelemetry/app/app.py#L69
Will the instrumentor auto-generate random spans?

[doc]: https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/requests/requests.html

Reply:
It will generate a span but it won't be random. No harm in having this extra span imo